packages/wallets/src/features/cfd/flows/MT5/AvailableMT5AccountsList/AvailableMT5AccountsList.tsx

```diff
@@ -1,34 +1,81 @@
 import { THooks } from '../../../../../types';
-import { getMarketTypeDetails } from '../../../constants';
+import { getMarketTypeDetails, PRODUCT } from '../../../constants';
 import { JurisdictionModal, MT5PasswordModal } from '../../../modals';
 import './AvailableMT5AccountsList.scss';
 
+const LazyVerification = lazy(
+    () => import(/* webpackChunkName: "wallets-client-verification" */ '../../ClientVerification/ClientVerification')
+);
+
 type TProps = {
-    account: THooks.SortedMT5Accounts;
+    account: THooks.AvailableMT5Accounts;
 };
 
 const AvailableMT5AccountsList: React.FC<TProps> = ({ account }) => {
     const { data: activeWallet } = useActiveWalletAccount();
     const { setModalState, show } = useModal();
-    const { description, title } = getMarketTypeDetails()[account.market_type || 'all'];
+    const { description, title } = getMarketTypeDetails(account.product)[account.market_type || 'all'];
+    const [showMt5PasswordModal, setShowMt5PasswordModal] = useState(false);
 
     const onButtonClick = useCallback(() => {
-        activeWallet?.is_virtual
-            ? show(<MT5PasswordModal marketType={account?.market_type || 'synthetic'} platform={account.platform} />)
-            : show(<JurisdictionModal />);
+        if (activeWallet?.is_virtual || account.product === PRODUCT.SWAPFREE) {
+            show(
+                <MT5PasswordModal
+                    isVirtual={activeWallet?.is_virtual}
+                    marketType={account?.market_type || 'synthetic'}
+                    platform={account.platform}
+                    product={account.product}
+                />
+            );
+        } else if (account.product === PRODUCT.ZEROSPREAD) {
+            show(
+                <LazyVerification
+                    onCompletion={() => {
+                        setShowMt5PasswordModal(true);
+                    }}
+                    selectedJurisdiction={account.shortcode}
+                />
+            );
+        } else {
+            show(<JurisdictionModal />);
+        }
         setModalState('marketType', account.market_type);
-    }, [activeWallet?.is_virtual, show, account.market_type, account.platform, setModalState]);
+        setModalState('selectedJurisdiction', account.shortcode);
+    }, [
+        activeWallet?.is_virtual,
+        show,
+        account.market_type,
+        account.platform,
+        setModalState,
+        account.product,
+        account?.shortcode,
+    ]);
+
+    useEffect(() => {
+        if (showMt5PasswordModal) {
+            show(
+                <MT5PasswordModal
+                    isVirtual={activeWallet?.is_virtual}
+                    marketType={account?.market_type || 'all'}
+                    platform={account.platform}
+                    product={account.product}
+                />
+            );
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [showMt5PasswordModal]);
```

Comment:
Why is the `useEffect` needed @hasan-deriv, I think just the conditional is enough right? 
```suggestion
        if (showMt5PasswordModal) {
            show(
                <MT5PasswordModal
                    isVirtual={activeWallet?.is_virtual}
                    marketType={account?.market_type || 'all'}
                    platform={account.platform}
                    product={account.product}
                />
            );
        }
```

Reply:
Actually @hasan-deriv, I see, it's needed, because it has the "side effect" of accessing the DOM ("modal_root").  But I think it's better to instead make this a function, and then call it here instead of updating state using the `setShowMt5PasswordModal`.

````typescript

        const showMt5PasswordModal = () => {
        show(
                        <MT5PasswordModal
                            isVirtual={activeWallet?.is_virtual}
                            marketType={account?.market_type || 'all'}
                            platform={account.platform}
                            product={account.product}
                        />
                    );
        }


        <LazyVerification
                    onCompletion={() => {
                        showMt5PasswordModal();
                    }}
                    selectedJurisdiction={account.shortcode}
                />